2.4/root/usr/share/container-scripts/mongodb/common.sh

```diff
@@ -244,7 +244,16 @@ function setup_keyfile() {
     exit 0
```

Comment:
Exiting really depends on the caller... if the call is in a subshell, it will exit the subshell.
In the case of setup_keyfile, I see it is called in the top level, I totally agree this `exit 0` is suspicious.

And so are the `exit 1`. They should probably be `return 1`, and exiting would be caused by the `errexit` shell option that we have set.